core/variant.cpp

```diff
@@ -2687,7 +2687,7 @@ uint32_t Variant::hash() const {
 		} break;
 		case INT: {
 
-			return _data._int;
+			return hash_one_uint64(uint64_t(_data._int));
```

Comment:
I'd prefer this move to a dedicated PR, as I fear it's breaking compatibility. The other changes can be merged for 3.2, but this one I doubt.

Reply:
See also: https://github.com/godotengine/godot/issues/33581